fh-ios-sdk/Sync/FHSyncDataset.m

```diff
@@ -750,6 +738,41 @@ - (void)syncRecordsSuccess:(NSDictionary *)resData {
     [self syncCompleteWithCode:@"online"];
 }
 
+/* 
+ If the records returned from syncRecord request contains elements in pendings,
+ it means there are local changes that haven't been applied to the cloud yet.
+ Remove those records from the response to make sure local data will not be
+ overridden (blinking desappear / reappear effect).
+ */
+- (void)applyPendingChangesToRecords:(NSMutableDictionary *)resData {
+    DLog(@"SyncRecords result = %@ pending = %@", resData, self.pendingDataRecords);
+    [self.pendingDataRecords enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+        FHSyncPendingDataRecord* pendingRecord = (FHSyncPendingDataRecord*)obj;
+        NSMutableDictionary* resRecord = nil;
+        if (resData[@"create"]) {
+            resRecord = resData[@"create"];
+            if (resRecord && resRecord[pendingRecord.uid]) {
+                pendingRecord.preData.data = resRecord[pendingRecord.uid][@"data"];
```

Comment:
I think this will not update the hash value of the preData, it will end up with the data of preData is the new one from the response, but the hash value is still the old one. You can do it this way:

```objc
FHSyncDataRecord* newpre = [[FHSyncDataRecord alloc] initWithData: resRecord[pendingRecord.uid]];
pendingRecord.preData = newpre;
```